src/gt4py/backend/module_generator.py

```diff
@@ -110,28 +111,29 @@ def make_args_data_from_gtir(pipeline: GtirPipeline) -> ModuleData:
     referenced_field_params = {
         param.name for param in node.params if isinstance(param, gtir.FieldDecl)
     }
-    for name in referenced_field_params:
+    for name in sorted(referenced_field_params):
         data.field_info[name] = FieldInfo(
             access=AccessKind.READ_WRITE if name in write_fields else AccessKind.READ_ONLY,
             boundary=field_extents[name].to_boundary(),
-            axes=list(dimension_flags_to_names(node.symtable_[name].dimensions).upper()),
+            axes=tuple(dimension_flags_to_names(node.symtable_[name].dimensions).upper()),
+            data_dims=tuple(node.symtable_[name].data_dims),
             dtype=numpy.dtype(node.symtable_[name].dtype.name.lower()),
         )
 
     referenced_scalar_params = set(node.param_names).difference(referenced_field_params)
-    for name in referenced_scalar_params:
+    for name in sorted(referenced_scalar_params):
         data.parameter_info[name] = ParameterInfo(
             dtype=numpy.dtype(node.symtable_[name].dtype.name.lower())
         )
 
     unref_params = get_unused_params_from_gtir(pipeline)
-    for param in unref_params:
+    for param in sorted(unref_params, key=lambda decl: decl.name):
         if isinstance(param, gtir.FieldDecl):
             data.field_info[param.name] = None
         elif isinstance(param, gtir.ScalarDecl):
             data.parameter_info[param.name] = None
 
-    data.unreferenced = {param.name for param in unref_params}
+    data.unreferenced = {*sorted(param.name for param in unref_params)}
```

Comment:
Even here: `data.unreferenced` is a set, so if I'm understanding correctly the sorted does nothing here.

Reply:
Oops, you're completely right here, I didn't realized that `unreferenced` had been migrated to a `set`. I've converted it back to a list.